pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -219,12 +254,194 @@ func (r *ReconcileNodeMaintenance) stopNodeMaintenance(nodeName string) error {
 		return err
 	}
 
+	if err := r.invalidateLease(nodeName); err != nil {
+		return err
+    }
+
+	return nil
+}
+
+func (r *ReconcileNodeMaintenance) invalidateLease(nodeName string) error {
+		if r.isLeaseSupported {
+				log.Info("Lease object supported,  invalidating lease")
+
+				nName := apitypes.NamespacedName{Namespace: LeaseNamespace, Name: nodeName}
+				lease := &coordv1beta1.Lease{}
+
+				if err := r.client.Get(context.TODO(), nName, lease); err != nil {
+
+					if  errors.IsResourceExpired(err) {
+						return nil
+					}
+					return err
+				}
+				lease.Spec.AcquireTime = nil
+				lease.Spec.LeaseDurationSeconds = nil;
+				lease.Spec.RenewTime = nil
+				lease.Spec.LeaseTransitions = nil
+
+				if err := r.client.Update(context.TODO(), lease); err != nil {
+					return err
```

Comment:
Probably some errors we need to swallow here, including if the lease was deleted from under us

Reply:
the error is propagated via  stopNodeMaintenance  to Reconciles, here the errors.IsNotFound conditions is ignored.